packages/typescript-estree/src/CachingMap.ts

```diff
@@ -0,0 +1,23 @@
+export class CachingMap<Key, Value> {
```

Comment:
I've seen this pattern at least twice now: once in the `getProjectConfigFiles` in this PR, and once in `parser.ts` > `existingPrograms`. Are there other places in the repo that could use it? I did a search for `.get(` and `.set(` (which found `existingPrograms`) but didn't see any others.

Reply:
Meh, ended up not using this.